### PR TITLE
#19 Add Password Confirmation Box

### PR DESCRIPTION
**Pull request by mariezimmy**
Updated RegisterServlet to include password and confirm password match logic, updated register.jsp to include confirm password box, and updated RegisterServletTest to include testing the logic added in RegisterServlet

### CODE REVIEW

**Comment by mariezimmy** [comment]
Not quite sure what to do about the ci/circleci test failure. @fatlotus any advice? I haven't downloaded/intalled the linter, should I start with that?

**Comment by fatlotus** [comment]
Yeah, give that a go. I'll take a look this afternoon and see if I can have it print out more useful output (which lines need to be reformatted, for example).

File: src/main/java/codeu/controller/RegisterServlet.java [hunk 1]
```diff
@@ -47,13 +47,20 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
...
     String password = request.getParameter("password");
+    String confirmPassword = request.getParameter("confirmPassword");
 
     if (!username.matches("[\\w*\\s*]*")) {
       request.setAttribute("error", "Please enter only letters, numbers, and spaces.");
       request.getRequestDispatcher("/WEB-INF/view/register.jsp").forward(request, response);
       return;
     }
 
+    if (password!= null && confirmPassword!= null &&!password.equals(confirmPassword)) {
```
Comment (fatlotus) [inline src/main/java/codeu/controller/RegisterServlet.java]: I am not a linter, but is suspect it is unhappy about the lack of spaces before the !s.

File: src/test/java/codeu/controller/RegisterServletTest.java [hunk 2]
```diff
@@ -87,4 +87,17 @@ public void testDoPost_ExistingUser() throws IOException, ServletException {
...
+  @Test
+  public void testDoPost_InvalidConfirmPassword() throws IOException, ServletException {
+    Mockito.when(mockRequest.getParameter("username")).thenReturn("test username");
+    Mockito.when(mockRequest.getParameter("password")).thenReturn("test password");
+    Mockito.when(mockRequest.getParameter("confirmPassword")).thenReturn("test confirm password");
+
+    registerServlet.doPost(mockRequest, mockResponse);
+
+    Mockito.verify(mockRequest)
+            .setAttribute("error", "Your password and confirmation password do not match.");
```
Comment (fatlotus) [inline src/test/java/codeu/controller/RegisterServletTest.java]: Probably has opinions here too.

Comment (fatlotus) [inline src/test/java/codeu/controller/RegisterServletTest.java]: Also, do you need to change the `_NewUser()` case?

File: src/main/java/codeu/controller/RegisterServlet.java [hunk 3]
```diff
@@ -47,13 +47,20 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
...
     String password = request.getParameter("password");
+    String confirmPassword = request.getParameter("confirmPassword");
 
     if (!username.matches("[\\w*\\s*]*")) {
       request.setAttribute("error", "Please enter only letters, numbers, and spaces.");
       request.getRequestDispatcher("/WEB-INF/view/register.jsp").forward(request, response);
       return;
     }
 
+    if (password != null && confirmPassword != null && !password.equals(confirmPassword)) {
```
Comment (fatlotus) [inline src/main/java/codeu/controller/RegisterServlet.java]: Presumably we also need to validate that the user has entered a confirmation password, right?

Comment (mariezimmy) [reply src/main/java/codeu/controller/RegisterServlet.java]: Yes! I'll get on that. Thanks!

File: src/main/java/codeu/controller/RegisterServlet.java [hunk 4]
```diff
@@ -47,13 +47,26 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
...
       return;
     }
 
+    if (confirmPassword == null) {
+      request.setAttribute("error", "Please enter a confirmation password.");
+      request.getRequestDispatcher("/WEB-INF/view/register.jsp").forward(request, response);
+      return;
+    }
+
+    if (password != null && !password.equals(confirmPassword)) {
```
Comment (fatlotus) [inline src/main/java/codeu/controller/RegisterServlet.java]: Password too, right?

Comment (fatlotus) [reply src/main/java/codeu/controller/RegisterServlet.java]: Ah, or we just go ahead and merge this and let the password validation catch it.

Comment (mariezimmy) [reply src/main/java/codeu/controller/RegisterServlet.java]: That sounds good! I'll go ahead and merge!

**Pull request merged**